tests/ludwig/modules/test_convolutional_modules.py

```diff
@@ -1,13 +1,291 @@
 import pytest
 from typing import Callable, Dict, List, Optional, Tuple, Union
-
+import numpy as np
 import torch
 
-from ludwig.modules.convolutional_modules import Conv2DLayer, Conv2DStack,\
-    Conv2DLayerFixedPadding, ResNetBlock, ResNetBottleneckBlock,\
-    ResNetBlockLayer, ResNet
+from ludwig.utils.image_utils import get_img_output_shape
+from ludwig.modules.convolutional_modules import Conv2DLayer, Conv2DStack, \
+    Conv2DLayerFixedPadding, ResNetBlock, ResNetBottleneckBlock, \
+    ResNetBlockLayer, ResNet, Conv1DLayer, Conv1DStack, ParallelConv1DStack, \
+    ParallelConv1D
+
+BATCH_SIZE = 2
+SEQ_SIZE = 50
+HIDDEN_SIZE = 64
+NUM_FILTERS = 32
```

Comment:
Let's make them a bit smaller, for instance:
  SEQ_SIZE = 10
  HIDDEN_SIZE = 8
  NUM_FILTERS = 4
to speed up tests

Reply:
I can make the change but then I'll have to reduce the combinations tested.  For example, with seq_size=10, will fail with the larger values in this set of parameters:
```
@pytest.mark.parametrize('pool_size, pool_padding, pool_stride',
                         [(None, None, None), (3, 'same', 1), (5, 'same', 1),
                          (3, 'valid', 2), (5, 'valid', 2)])
```
Any recommendations on the range of values to test for?

Reply:
Here are some of the errors that occur with the smaller sizes:
```
>       return torch.max_pool1d(input, kernel_size, stride, padding, dilation, ceil_mode)
E       RuntimeError: Given input size: (4x1x4). Calculated output size: (4x1x0). Output size is too small
E       RuntimeError: Calculated padded input size per channel: (10). Kernel size: (13). Kernel size can't be greater than actual input size
E       RuntimeError: Given input size: (4x1x2). Calculated output size: (4x1x-1). Output size is too small
```

Reply:
In terms for speed the `test_conv1d_layer` is the one most affected by the change in parameters.  With the sizing that is currently inlace, before your recommendations, 360 tests are generated and it takes about 2 seconds on my laptop.

Reply:
@w4nderlust   made the adjustments per comments except for seq_size.  The minimum value required to allow testing some of the combinations of parameter is 17.  I did cut back on some of combinations.  So the number of generated test cases is about half.  From what I can tell, this gives the minimum combinations that will exercise all the code paths.